### PR TITLE
TLS support for PostgreSQL managed by VSHN

### PR DESCRIPTION
**Pull request by wejdross**
## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [ ] Update the documentation.
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [x] Link this PR to related issues or PRs.

### CODE REVIEW

File: component/vshn_postgres.jsonnet [hunk 1]
```diff
@@ -122,13 +184,24 @@ local composition =
                                   [
                                     {
                                       sgObjectStorage: '',
-                                      cronSchedule: '',
+                                      cronSchedule: '0 22 * * *',
```
Comment (Kidswiss) [inline component/vshn_postgres.jsonnet:187]: Is this change intended? It looks like it's unrelated to the TLS configuration?

**Comment by wejdross** [comment]
```
apiVersion: v1
kind: Namespace
metadata:
  name: development
  labels:
    name: development
---
apiVersion: vshn.appcat.vshn.io/v1
kind: VSHNPostgreSQL
metadata:
  name: my-postgres-example
  namespace: development
spec:
  parameters:
    service:
      majorVersion: "15"
    backup:
      schedule: '* * * * *'
  writeConnectionSecretToRef:
    name: postgres-creds-connection
---
apiVersion: v1
kind: Pod
metadata:
  name: postgres-client
  namespace: development
spec:
  containers:
  - name: postgres
    image: postgres:15
    resources:
      limits:
        memory: "128Mi"
        cpu: "500m"
    command:
      - /bin/sh
      - -c
    args:
      - PGPASSWORD=$POSTGRESQL_PASSWORD psql "sslmode=require host=$POSTGRESQL_HOST port=$POSTGRESQL_PORT dbname=$POSTGRESQL_DB" -U $POSTGRESQL_USER -l
    envFrom:
      - secretRef:
          name: postgres-creds-connection
  restartPolicy: Never
  ```

example claim.yaml to tests

File: component/vshn_postgres.jsonnet [hunk 2]
```diff
@@ -330,6 +452,25 @@ local composition =
             // add backup configuration
           ],
         },
+        // {
```
Comment (Kidswiss) [inline component/vshn_postgres.jsonnet:455]: You can probably remove the unused code block.

File: component/vshn_postgres.jsonnet [hunk 3]
```diff
@@ -54,7 +58,65 @@ local composition =
...
+                                subject: {
+                                  organizations: [
+                                    'vshn-appcat',
+                                  ],
+                                },
+                                isCA: false,
+                                privateKey: {
+                                  algorithm: 'RSA',
+                                  encoding: 'PKCS1',
+                                  size: 2048,
```
Comment (Kidswiss) [inline component/vshn_postgres.jsonnet:101]: I'd probably go with 4096 instead of 2048 for RSA.

**Review by Kidswiss — CHANGES_REQUESTED** [review]
A few smaller things I found.

File: component/vshn_postgres.jsonnet (hunk 1 above)
Comment (wejdross) [reply component/vshn_postgres.jsonnet:187]: this fixes issues with `default` in apis/vshn/v1/dbaas_vshn_postgresql.go because it's not set correctly, so I moved it here, it's harmless change which actually double ensure that correct schedule is applied to backup job

File: component/vshn_postgres.jsonnet (hunk 2 above)
Comment (wejdross) [reply component/vshn_postgres.jsonnet:455]: ok

File: component/vshn_postgres.jsonnet (hunk 3 above)
Comment (wejdross) [reply component/vshn_postgres.jsonnet:101]: ok

File: component/vshn_postgres.jsonnet (hunk 1 above)
Comment (Kidswiss) [reply component/vshn_postgres.jsonnet:187]: I'd probably put it in a separate PR then, as it's not related to TLS at all. So that things are clean.

File: component/vshn_postgres.jsonnet (hunk 3 above)
Comment (wejdross) [reply component/vshn_postgres.jsonnet]: fixed
